agent/taskresource/envFiles/envfile.go

```diff
@@ -322,6 +323,19 @@ func (envfile *EnvironmentFileResource) Create() error {
 	return nil
 }
 
+// createEnvfileDirectory creates the directory that we will be writing the
+// envfile to - needs to be called for each different envfile
+func (envfile *EnvironmentFileResource) createEnvfileDirectory(bucket, key string) error {
+	// create directories to include bucket and key but not the actual resulting file
+	envfileDir := filepath.Join(envfile.resourceDir, bucket, strings.TrimRight(key, string(filepath.Separator)))
```

Comment:
we should use filepath.Dir instead of strings.TrimRight.

Reply:
+1